theano/sandbox/gpuarray/type.py

```diff
@@ -120,7 +190,7 @@ def filter_variable(self, other, allow_convert=True):
                                  str(self.broadcastable)))
```

Comment:
I do this for now since convert_variable repeats a lot of the checks that filter_variable does.  In order to avoid paying that cost most of the time, I only call it when I know it could be needed.

But this could be refactored to call convert_variable() when allow_convert is True and make all the checks otherwise.  I'd like to make that a ticket and not part of this PR though.